app/models/goal.py

```diff
@@ -1,5 +1,33 @@
 from app import db
-
+from app.models.task import Task
 
 class Goal(db.Model):
+
     goal_id = db.Column(db.Integer, primary_key=True)
+    title = db.Column(db.String)
+    tasks = db.relationship('Task', backref='goal', lazy=True)
+
+
+    def to_dict(self):
+        goal_as_dict={}
+        
+        goal_as_dict["id"] = self.goal_id
+        goal_as_dict["title"] = self.title
+        return goal_as_dict
+    
+    def to_dict_with_tasks(self):
+
+        goal_as_dict = self.to_dict()
+        tasks_dicts = []
+        for task in self.tasks:
+            tasks_dicts.append(task.to_dict())
+
+        goal_as_dict["tasks"] = tasks_dicts
+
+        return goal_as_dict
```

Comment:
Similar to above, I think there are ways to refactor this. One way might be:

```python
        goal = self.to_dict()
        tasks = []
        for task in self.tasks:
            tasks.append(task.to_dict())
        goal["tasks"] = tasks
        return goal
```

In my suggestion, I changed the variable names to `goal` and `tasks` because I believe contextually the data structure is obvious